vault/core.go

```diff
@@ -648,15 +654,15 @@ func NewCore(conf *CoreConfig) (*Core, error) {
 	c.reloadFuncsLock.Unlock()
 	conf.ReloadFuncs = &c.reloadFuncs
 
-	// Setup the backends
 	logicalBackends := make(map[string]logical.Factory)
 	for k, f := range conf.LogicalBackends {
 		logicalBackends[k] = f
 	}
-	_, ok = logicalBackends["kv"]
+	_, ok := logicalBackends["kv"]
```

Comment:
I think we should do the same thing here with `plugin`. It's mostly helpful for tests, but we should ensure the logicalBackends and credentialBackends maps always have the plugin backend in it